injected/src/features/autofill-password-import.js

```diff
@@ -76,39 +99,116 @@ export default class AutofillPasswordImport extends ContentFeature {
                 ? {
                     style: this.signInButtonStyle,
                     element,
-                    shouldTap: this.#signInButtonSettings?.shouldAutotap ?? false
+                    shouldTap: this.#signInButtonSettings?.shouldAutotap ?? false,
+                    shouldWatchForRemoval: false
                 }
                 : null
         } else {
             return null
         }
     }
 
+    /**
+     * Removes the overlay if it exists.
+     */
+    removeOverlayIfNeeded () {
+        const existingOverlay = document.getElementById(OVERLAY_ID)
+        if (existingOverlay != null) {
+            existingOverlay.style.display = 'none'
+            existingOverlay.remove()
+        }
+    }
+
+    /**
+     * Inserts an overlay element to animate, by adding a div to the body 
+     * and styling it based on the found element. 
+     * @param {HTMLElement|Element} mainElement 
+     * @param {any} style 
+     * @returns {HTMLElement|Element|null}
+     */
+    insertOverlayElement (mainElement, style) {
+        this.removeOverlayIfNeeded()
+
+        const overlay = document.createElement('div')
+        overlay.setAttribute('id', OVERLAY_ID)
+        const svgElement = mainElement.parentNode?.querySelector('svg') ?? mainElement.querySelector('svg')
+
+        const isRound = style.borderRadius === '100%'
+        const elementToCenterOn = isRound ? svgElement : mainElement
+        if (elementToCenterOn) {
+            const { top, left, width, height } = elementToCenterOn.getBoundingClientRect()
+            overlay.style.position = 'absolute'
+    
+            overlay.style.top = `calc(${top}px + ${window.scrollY}px - ${isRound ? height / 2 : 0}px - 1px - ${style.offsetTopEm}em)`
```

Comment:
I'd prefer these ternaries as variables outside of the template string to simplify the expression a little. Feel free to skip if you care about keeping.